toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -493,6 +495,22 @@ func PopulateInstallRoot(installChroot *safechroot.Chroot, packagesToInstall []s
 		return
 	}
 
+	// imageconfigvalidator should have ensured that we intend to install shadow-utils, so we can go ahead and do that here.
+	if len(config.Users) > 0 || len(config.Groups) > 0 {
+		if !sliceutils.ContainsValue(packagesToInstall, "shadow-utils") {
+			err = fmt.Errorf("shadow-utils package is required when setting users or groups")
```

Comment:
I'd mention that `shado-utils` must be listed explicitly. Otherwise, it's going to be confusing, if someone adds a package, which pulls in `shadow-utils`.